src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
```

Comment:
I think you meant regardless of how high threshold is. It makes sense. I believe browsers don’t support more than 60 fps right now (Chrome gave 60 or slightly less on empty pages when we tried this out) but we should be future proof.